pkg/flowcli/project/transaction.go

```diff
@@ -0,0 +1,243 @@
+)
+
+func NewTransaction() *Transaction {
+	return &Transaction{
+		tx:        flow.NewTransaction(),
+		contracts: []templates.Contract{},
+	}
+}
+
+type Transaction struct {
+	signer    *Account
+	role      signerRole
+	proposer  *Account
+	payer     flow.Address
+	tx        *flow.Transaction
+	contracts []templates.Contract
+}
+
+func (t *Transaction) Signer() *Account {
+	return t.signer
+}
+
+func (t *Transaction) FlowTransaction() *flow.Transaction {
+	return t.tx
+}
+
+func (t *Transaction) SetPayloadFromFile(filename string) error {
+	partialTxHex, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return fmt.Errorf("failed to read partial transaction from %s: %v", filename, err)
+	}
+	partialTxBytes, err := hex.DecodeString(string(partialTxHex))
+	if err != nil {
+		return fmt.Errorf("failed to decode partial transaction from %s: %v", filename, err)
+	}
+	tx, err := flow.DecodeTransaction(partialTxBytes)
+	if err != nil {
+		return fmt.Errorf("failed to decode transaction from %s: %v", filename, err)
+	}
+
+	t.tx = tx
+	return nil
+}
+
+func (t *Transaction) SetScriptWithArgsFromFile(filepath string, args []string, argsJSON string) error {
+	script, err := util.LoadFile(filepath)
+	if err != nil {
+		return err
+	}
+
+	return t.SetScriptWithArgs(script, args, argsJSON)
+}
+
+func (t *Transaction) SetScriptWithArgs(script []byte, args []string, argsJSON string) error {
+	t.tx.SetScript(script)
+	return t.AddRawArguments(args, argsJSON)
+}
+
+func (t *Transaction) SetSigner(account *Account) error {
+	err := account.ValidateKey()
+	if err != nil {
+		return err
+	}
+
+	t.signer = account
+	return nil
+}
+
+func (t *Transaction) SetProposer(account *Account) error {
+	err := account.ValidateKey()
+	if err != nil {
+		return err
+	}
+
+	t.proposer = account
+	return nil
+}
+
+func (t *Transaction) SetPayer(address flow.Address) {
+	t.payer = address
+}
+
+func (t *Transaction) AddRawArguments(args []string, argsJSON string) error {
+	txArguments, err := flowcli.ParseArguments(args, argsJSON)
+	if err != nil {
+		return err
+	}
+
+	return t.AddArguments(txArguments)
+}
+
+func (t *Transaction) AddArguments(args []cadence.Value) error {
+	for _, arg := range args {
+		err := t.AddArgument(arg)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *Transaction) AddArgument(arg cadence.Value) error {
+	return t.tx.AddArgument(arg)
+}
+
+func (t *Transaction) AddAuthorizers(addresses []string) error {
+	for _, address := range addresses {
+		err := t.AddAuthorizer(address)
+		if err != nil { // return error even if one breaks
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *Transaction) AddAuthorizer(address string) error {
+	authorizerAddress := flow.HexToAddress(address)
+	if authorizerAddress == flow.EmptyAddress {
+		return fmt.Errorf("invalid authorizer address provided %s", address)
+	}
+
+	t.tx.AddAuthorizer(authorizerAddress)
+	return nil
+}
+
+func (t *Transaction) SignerRole(role string) error {
```

Comment:
What's the purpose of this function?

Reply:
Its purpose is to set the signer role which is later used, it's something I just moved from the previous implementation.